sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt

```diff
@@ -0,0 +1,253 @@
+package com.mapbox.maps.viewannotation
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewTreeObserver
+import android.widget.FrameLayout
+import androidx.annotation.LayoutRes
+import androidx.asynclayoutinflater.view.AsyncLayoutInflater
+import com.mapbox.maps.*
+import com.mapbox.maps.plugin.ViewPlugin
+import java.util.concurrent.ConcurrentHashMap
+
+internal class ViewAnnotationManagerImpl(
+  private val mapView: MapView
+) : ViewAnnotationManager, ViewAnnotationPositionsListener {
+
+  private val mapboxMap: MapboxMap = mapView.getMapboxMap()
+
+  init {
+    mapView.requestDisallowInterceptTouchEvent(false)
+    mapboxMap.setViewAnnotationPositionsUpdateListener(this)
+  }
+
+  private val asyncInflater by lazy {
+    try {
+      AsyncLayoutInflater(mapView.context)
+    } catch (e: NoClassDefFoundError) {
+      throw RuntimeException(
+        "Please add https://mvnrepository.com/artifact/androidx.asynclayoutinflater/asynclayoutinflater/1.0.0 dependency " +
+          "to your project to make use of asynchronous view inflation when adding view annotation!"
+      )
+    }
+  }
+
+  private val annotationMap = ConcurrentHashMap<String, ViewAnnotation>()
+  private val idLookupMap = ConcurrentHashMap<View, String>()
+
+  // structs needed for drawing, declare them only once
+  private val currentViewsDrawnMap = HashMap<String, ScreenCoordinate>()
+  private val positionDescriptorMap = HashMap<String, ViewPosition>()
+  private val idsToRepositionSet = HashSet<String>()
+  private val idsToAddSet = HashSet<String>()
+  private val idsToDeleteSet = HashSet<String>()
+
+  override fun addViewAnnotation(
+    @LayoutRes resId: Int,
+    options: ViewAnnotationOptions,
+    asyncInflateCallback: (View) -> Unit
+  ) {
+    validateOptions(options)
+    asyncInflater.inflate(resId, mapView) { view, _, _ ->
+      asyncInflateCallback.invoke(prepareViewAnnotation(view, options))
+    }
+  }
+
+  override fun addViewAnnotation(
+    @LayoutRes resId: Int,
+    options: ViewAnnotationOptions
+  ): View {
+    validateOptions(options)
+    val view = LayoutInflater.from(mapView.context).inflate(resId, mapView, false)
+    return prepareViewAnnotation(view, options)
+  }
+
+  override fun addViewAnnotation(view: View, options: ViewAnnotationOptions) {
+    if (idLookupMap.contains(view)) {
+      throw RuntimeException(
+        "Trying to add view annotation that was already added before! " +
+          "Please consider deleting annotation view ($view) beforehand."
+      )
+    }
+    validateOptions(options)
+    prepareViewAnnotation(view, options)
+  }
+
+  override fun removeViewAnnotation(view: View): Boolean {
+    val id = idLookupMap[view] ?: return false
```

Comment:
`remove` method will return the previous value associated with key, or null if there was no mapping for key,
so here we can call remove directly
`val id = idLookupMap.remove(view)`

Reply:
thanks